package StatisticsModule_Proto;

option java_package = "zsdn.protocol";
option java_outer_classname = "StatisticsModuleProtocol";

// Main message type for messages FROM StatisticsModule

message From {

    oneof FromMsg {
        SwitchStats stats = 1;
    }
	
    message SwitchStats {
	required uint64 uniqueID = 1;
	optional PortStatslist portStats = 2;
	optional TableStatslist tableStats = 3;
    }	

    message PortStatslist {
	repeated SinglePortStatistics portStats = 1;
    }

    message TableStatslist {
	repeated SingleTableStatistics tableStats = 1;
    }

    message SinglePortStatistics {
	required uint32 portNumber = 1; 
	optional uint64 numberOfPackagesReceived = 2;
        optional uint64 numberOfPackagesSent = 3;
        optional uint64 numberOfPackagesDropedByReceiver = 4;
        optional uint64 numberOfPackagesDropedBySender = 5;
        optional uint64 numberOfCollisions = 6;
        optional uint64 numberOfErrorsFromSender = 7;
        optional uint64 numberOfErrorsFromReceiver = 8;
	optional uint64 numberOfBytesReceived = 9;
        optional uint64 numberOfBytesSent = 10;
    }

    message SingleTableStatistics {
	required uint32 tableID = 1;
        optional uint32 activeEntries = 2;
        optional uint64 PackagesLookedUp = 3;
        optional uint64 PackagesHittingTheTable = 4;
	optional uint32 maxEntries = 5;
    }
}
